src/engine/source/vdscanner/src/osScanner.hpp

```diff
@@ -56,7 +56,8 @@ class TOsScanner final : public utils::patterns::AbstractHandler<std::shared_ptr
 
         const auto osCPE = ScannerHelper::parseCPE(data->osCPEName(m_databaseFeedManager->cpeMappings()).data());
 
-        auto vulnerabilityScan = [&](const std::string& cnaName,
+        auto vulnerabilityScan = [&, functionName = logging::getLambdaName(__FUNCTION__, "vulnerabilityScan").c_str()](
```

Comment:
We are using `c_str()` over an rvalue, which is a temporary object, and it will be destroyed at the end of the statement, the pointer returned by `c_str()` will become invalid. This leads to undefined behavior when you attempt to use `functionName` inside the lambda.

```suggestion
        auto vulnerabilityScan = [&, functionName = logging::getLambdaName(__FUNCTION__, "vulnerabilityScan")](
```

After removing c_str():
```
2024-08-29 14:18:08.923 131338:131378 packageScanner.hpp:92 at scanPackageTranslation::<lambda>scanPackage(): debug: Initiating a vulnerability scan for package 'adobe photoshop 2024' (win) (adobe inc.) with CVE Numbering Authorities (CNA) 'nvd' on Agent '' (ID: '002', Version: '').
```